### PR TITLE
issue-709 : added a method for scrolling documents

### LINKED ISSUE #709: ElasticsearchService cannot fetch 10 001st record
- from and size parameters only work for up to 10k records
- investigate possible fixes (search_after, PIT)

### PR DESCRIPTION
**Pull request by DanaSunal**
Fixes #709

### CODE REVIEW

File: server/applicationPattern/applicationPattern/services/ElasticsearchService.js [hunk 1]
```diff
@@ -265,6 +265,95 @@ class ElasticsearchService {
...
+      size: 10000,
+      body: {
+        query: query
+      }
+    });
+    let total = response.body.hits.total.value;
+
+    // check if we can return the number of results desired by input parameters
+    if (from > total) {
+      console.log("There are not enough results to satisfy input.");
```
[the remark this answers is not in the document]
Comment (DanaSunal) [reply server/applicationPattern/applicationPattern/services/ElasticsearchService.js]: Done.

File: server/applicationPattern/applicationPattern/services/ElasticsearchService.js [hunk 2]
```diff
@@ -265,6 +265,95 @@ class ElasticsearchService {
...
+      throw new Error("This method should not be used for queries with results below 10k.");
+    }
+    let indexAlias = await getIndexAliasAsync(uuid);
+    let client = await this.getClient(false, uuid);
+
+    // store here unprocessed results
+    let responseQueue = [];
+    // store here processed results
+    let results = [];
+    let took = 0;
```
[the remark this answers is not in the document]
Comment (DanaSunal) [reply server/applicationPattern/applicationPattern/services/ElasticsearchService.js]: Done.

**Review by MartinSunal — DISMISSED** [review]

File: server/applicationPattern/applicationPattern/services/ElasticsearchService.js [hunk 3]
```diff
@@ -265,6 +265,97 @@ class ElasticsearchService {
...
+   * @param {Number} from index from which we want to read documents
+   * @param {Number} size how many documents do we want to read
+   * @param {Object} query ES query to be used to filter the documents
+   * @param {String} [uuid] optional, UUID of ES client
+   * @throws Error if from+size do not exceed 10k
+   * @returns {Promise<Object>} { response, took } 
+   */
+  async scroll(from, size, query, uuid) {
+    // check if we should be using this method
+    if ((from + size) < 10000) {
```
Comment (MartinSunal) [inline server/applicationPattern/applicationPattern/services/ElasticsearchService.js:285]: `(from + size) < 10000` should be `(from + size) <= 10000`

Comment (DanaSunal) [reply server/applicationPattern/applicationPattern/services/ElasticsearchService.js]: Done.

**Review by MartinSunal — APPROVED** [review]

**Pull request merged**